bundles/binding/org.openhab.binding.insteonplm/src/main/java/org/openhab/binding/insteonplm/internal/driver/IOStream.java

```diff
@@ -23,18 +27,54 @@
  */
 
 public abstract class IOStream {
+	private static final Logger logger = LoggerFactory.getLogger(IOStream.class);
 	protected InputStream	m_in	= null;
 	protected OutputStream	m_out	= null;
+
 	/**
-	 * Getter for input stream from modem/hub
-	 * @return input stream for incoming data
+	 * read data from iostream
+	 * @param b byte array (output)
+	 * @param offset offset for placement into byte array
+	 * @param readSize size to read
+	 * @return number of bytes read
 	 */
-	public InputStream in() { return m_in; }
+	public int read(byte [] b, int offset, int readSize) {
+		int len = 0;
+		while (len < 1) {
+			try {
+				len = m_in.read(b, offset, readSize);
+			} catch (IOException e) {
+				logger.trace("got exception while reading: {}", e.getMessage());
+				while (!reconnect()) {
+					try {
+						logger.trace("sleeping before reconnecting");
+						Thread.sleep(10000);
+					} catch (InterruptedException ie) {
```

Comment:
the idea was to have a TRACE level statement in case of an error … so one can get an idea where the problem lies.